src/Registration/cReg/NiftiImageData3DDeformation.cpp

```diff
@@ -58,12 +59,17 @@ void NiftiImageData3DDeformation<dataType>::create_from_cpp(NiftiImageData3DTens
 {
     this->create_from_3D_image(ref);
 
-    reg_spline_getDeformationField(cpp.get_raw_nifti_sptr().get(),
-                                   this->_nifti_image.get(),
-                                   NULL,
-                                   false, //composition
-                                   true // bspline
-                                   );
+    // reg_spline_getDeformationField(cpp.get_raw_nifti_sptr().get(),
+    //                                this->_nifti_image.get(),
+    //                                NULL,
+    //                                false, //composition
+    //                                true // bspline
+    //                                );
+
+    reg_spline_getDefFieldFromVelocityGrid(cpp.get_raw_nifti_sptr().get(),
+                                           this->_nifti_image.get(),
+                                           false // the number of step is not automatically updated
+                                           );
```

Comment:
is this correct? `create_from_cpp` always call this?

Reply:
This isn't correct, this will only work with f3d2, I wrote it this way for speed of implementation. Really we should check to see if _use_velocity is set

Reply:
The problem being that _use_velocity is a member variable of NiftyF3dSym

Reply:
not sure what you mean, but maybe this just needs 2 different functions `create_from_cpp` and `create_from_velocity_field`. Or it needs some header inspection